### PR TITLE
switch from JSON to JSON3 dependency

### LINKED ISSUE #20: Use JSON3 instead of JSON dependency
Should hopefully be an allround improvement in both speed and inferred types.

cc @rafaqz (https://github.com/JuliaGeo/GeoInterface.jl/issues/23#issuecomment-500659441)

### PR DESCRIPTION
**Pull request by visr**
I tried to keep this change as minimal as possible.

Since JSON3 by default returns a `JSON3.Object` which is a `AbstractDict{Symbol, Any}`, I had to widen some signatures that before were restricted to `Dict{String, Any}`.

I did not want to change GeoInterface `FeatureCollection.CRS` and `Feature.properties` types, which are `Dict{String,Any}`. Therefore I had to convert to this type at the few places where these were created.

Otherwise, the `JSON3.Object` will behave similarly to a `Dict{String, Any}`, since it also supports `get` with `String` keys.

Fixes #20.

### CODE REVIEW

**Comment by visr** [comment]
I should note that in this way, this does not improve the package that much.

However, especially with #21 in mind, I believe JSON3 has the potential to simplify this package considerably, with struct mapping that allows us to parse the JSON directly to out structs. Then we can eliminate the current two staged approach in this package, with `GeoInterface concrete types ↔ Dict ↔ JSON`.

File: src/GeoJSON.jl [hunk 1]
```diff
@@ -20,16 +20,20 @@ julia> GeoJSON.parse("{\"type\": \"Point\", \"coordinates\": [30, 10]}")
 GeoInterface.Point([30.0, 10.0])
 ```
 """
-parse(input; kwargs...) = dict2geo(JSON.parse(input; kwargs...))
+parse(input) = dict2geo(JSON3.read(input))
```
Comment (yeesian) [inline src/GeoJSON.jl]: Can we rename this as `read(input)`, deprecate `GeoJSON.parse(input)`, and point people to `GeoJSON.read(input)` instead (for consistency with `JSON3`).

File: src/GeoJSON.jl [hunk 2]
```diff
@@ -48,11 +52,11 @@ function geojson end
 for geom in (:AbstractFeatureCollection, :AbstractGeometryCollection, :AbstractFeature,
         :AbstractMultiPolygon, :AbstractPolygon, :AbstractMultiLineString,
         :AbstractLineString, :AbstractMultiPoint, :AbstractPoint)
-    @eval geojson(obj::GeoInterface.$geom) = JSON.json(geo2dict(obj))
+    @eval geojson(obj::GeoInterface.$geom) = JSON3.write(geo2dict(obj))
```
Comment (yeesian) [inline src/GeoJSON.jl]: Can we rename this as `write(obj::GeoInterface.$geom)`, deprecate `GeoJSON.geojson(input)`, and point people to `GeoJSON.write(input)` instead (for consistency with `JSON3`).

**Comment by visr** [comment]
Great, thanks. Was considering `read` and `write` as well, will go ahead with that.

**Comment by visr** [comment]
I renamed the functions and added deprecations in the latest commit.

**Pull request merged**